### PR TITLE
FEATURE: holiday flair in sidebar

### PR DESCRIPTION
**Pull request by lis2**
Show flair when a user participating in direct message is on holiday

Screen Shot 2022-07-25 at 4 19 45 pm

### CODE REVIEW

File: assets/javascripts/discourse/initializers/chat-sidebar.js [hunk 1]
```diff
@@ -249,12 +250,21 @@ export default {
...
 
+            get oneOnOneMessage() {
+              return this.channel.chatable.users.length === 1;
+            }
+
             get text() {
-              return this.channel.title.replaceAll("@", "");
+              const username = this.channel.title.replaceAll("@", "");
+              if (this.oneOnOneMessage) {
+                return htmlSafe(`${username} ${decorateUsername(username)}`);
```
Comment (martin-brennan) [inline assets/javascripts/discourse/initializers/chat-sidebar.js:260]: Should `username` be here twice? Or does `decorateUsername(username)` not include the `username` in its output?

**Review by martin-brennan — APPROVED** [review]
Minor question but otherwise LGTM

File: assets/javascripts/discourse/initializers/chat-sidebar.js (hunk 1 above)
Comment (lis2) [reply assets/javascripts/discourse/initializers/chat-sidebar.js:260]: Yes, decorateUsername only contains suffix flairs. 

Example usage of decorate-username-selector is below. We display username and then suffix
https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/templates/user-selector-autocomplete.hbr#L7;L9

File: test/javascripts/acceptance/core-sidebar-test.js [hunk 2]
```diff
@@ -56,6 +59,21 @@ acceptance("Discourse Chat - Core Sidebar", function (needs) {
...
+        if (username === "hawk") {
+          return `<span class="on-holiday">${emojiUnescape(
+            ":desert_island:"
+          )}</span>`;
+        }
+      });
+    });
+  });
+  needs.hooks.afterEach(function () {
+    resetUsernameDecorators();
```
Comment (tgxworld) [inline test/javascripts/acceptance/core-sidebar-test.js:74]: I don't think we need to call this as this has been configured after each test; https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/tests/helpers/qunit-helpers.js#L168

**Review by tgxworld — APPROVED** [review]

**Pull request merged**